daemon/containerd/service.go

```diff
@@ -120,10 +119,18 @@ func (i *ImageService) StorageDriver() string {
 	return i.snapshotter
 }
 
-// ReleaseLayer releases a layer allowing it to be removed
+// ReleaseLayer releases the snapshot allowing it to be removed.
 // called from delete.go Daemon.cleanupContainer(), and Daemon.containerExport()
-func (i *ImageService) ReleaseLayer(rwlayer layer.RWLayer) error {
-	return errdefs.NotImplemented(errors.New("not implemented"))
+func (i *ImageService) ReleaseLayer(ctx context.Context, ctr *container.Container) error {
+	ls := i.client.LeasesService()
+	lease := leases.Lease{
+		ID: ctr.ID,
+	}
+	err := ls.Delete(ctx, lease, leases.SynchronousDelete)
+	if err != nil && cerrdefs.IsNotFound(err) {
+		return nil
+	}
+	return err
```

Comment:
I'm guessing the `err != nil` is even redundant here (but perhaps more readable this way 😅)

Alternatively, we could do the reverse, which would make the final return explicit that there's no error;

```suggestion
	if err != nil && !cerrdefs.IsNotFound(err) {
		return err
	}
	return nil
```